routes.py

```diff
@@ -136,6 +139,48 @@ def logout_page():
     return flask.render_template("logout.html",)
 
 
+@app.route('/profile')
+def profile():
+    i = current_user.id % 5
+    return flask.render_template("profile.html", currentuser=current_user, loginmanager=login_user, piclinks=piclinks[i])
+
+
+@app.route('/changepassword')
+def changepassword():
+    return flask.render_template("change-password.html")
+
+
+@app.route('/changepassword', methods=["POST"])
+def changepassword_post():
+    currentpassword = flask.request.form.get('currentpassword')
+    newpassword = flask.request.form.get('newpassword')
+    newpassword_confirm = flask.request.form.get('confirmnewpassword')
+    encrypt_changed_currentpassword = encodepassword(currentpassword)
+
+    validate_userpassword = UserCredential.query.filter_by(
+        email=current_user.email, password=str(encrypt_changed_currentpassword)).first()
+
+    # Check if email already in database
+    if not validate_userpassword:
+        flask.flash("The Current Password is invalid. Try something else!")
+        return flask.redirect(flask.url_for("changepassword"))
+
+    # Check if password and confirm does not matched
+    if newpassword != newpassword_confirm:
+        flask.flash(
+            "Your New Password did not match your Confirmed Password. Try again!")
+        return flask.redirect(flask.url_for("changepassword"))
+
```

Comment:
You could have a case where the new password = the old password and flash a message, but that's overkill for this point in the project lol. Cool feature! :)